pkg/agent/agent.go

```diff
@@ -262,42 +264,101 @@ func (a *Agent) Stop() {
 	}
 }
 
-func (a *Agent) hasLocalConfig() bool {
-	return a.configFile != ""
+var (
+	defaultConfig = &cfgapi.AgentConfig{
+		NodeResourceTopology: true,
+	}
+)
+
+func getAgentConfig(newConfig metav1.Object) *cfgapi.AgentConfig {
+	cfg := cfgapi.GetAgentConfig(newConfig)
+	if cfg == nil {
+		return defaultConfig
+	}
+	return cfg
 }
 
-func (a *Agent) setupClients() error {
+func (a *Agent) configure(newConfig metav1.Object) error {
 	if a.hasLocalConfig() {
+		log.Warn("running with local configuration, skipping cluster access client setup...")
 		return nil
 	}
 
-	cfg, err := a.getRESTConfig()
-	if err != nil {
-		return err
+	cfg := getAgentConfig(newConfig)
+
+	// We always need a HTTP/REST client and a K8s client.
+	if a.httpCli == nil {
+		log.Info("setting up HTTP/REST client...")
+		restCfg, err := a.getRESTConfig()
+		if err != nil {
+			return err
+		}
+
+		a.httpCli, err = rest.HTTPClientFor(restCfg)
+		if err != nil {
+			return fmt.Errorf("failed to setup kubernetes HTTP client: %w", err)
+		}
+
+		log.Info("setting up K8s client...")
+		a.k8sCli, err = k8sclient.NewForConfigAndClient(restCfg, a.httpCli)
+		if err != nil {
+			a.cleanupClients()
+			return fmt.Errorf("failed to setup kubernetes client: %w", err)
+		}
+
+		kubeCfg := *restCfg
+		err = a.cfgIf.SetKubeClient(a.httpCli, &kubeCfg)
+		if err != nil {
+			return fmt.Errorf("failed to setup kubernetes config resource client: %w", err)
+		}
 	}
 
-	a.httpCli, err = rest.HTTPClientFor(cfg)
-	if err != nil {
-		return fmt.Errorf("failed to setup kubernetes HTTP client: %w", err)
+	switch {
+	case cfg.PodResourceAPI && a.podResCli == nil:
```

Comment:
if / else ?

Reply:
well, maybe some comments might be added then, because for me this switch was less readable, until I figured out that it is also handling re-configure scenario where option can be enabled/disabled dynamically.